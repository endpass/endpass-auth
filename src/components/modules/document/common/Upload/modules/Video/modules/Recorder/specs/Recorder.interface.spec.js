import { createLocalVue, shallowMount } from '@vue/test-utils';
import PortalVue from 'portal-vue';
import setupI18n from '@/locales/i18nSetup';
import RecorderInterface from '../Recorder.interface';

const localVue = createLocalVue();
const i18n = setupI18n(localVue);
localVue.use(PortalVue);

describe('RecorderInterface', () => {
  let wrapper;

  const wrapperFactory = (options = {}) => {
    return shallowMount(RecorderInterface, {
      sync: false,
      localVue,
      i18n,
      ...options,
    });
  };

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = wrapperFactory();
  });

  describe('render', () => {
    it('should be a correct component', () => {
      expect(wrapper.isVueInstance()).toBe(true);
      expect(wrapper.name()).toEqual('RecorderInterface');
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('behavior', () => {
    it('should emit next', () => {
      expect.assertions(2);

      const recordedFile = 'recordedFile';

      expect(wrapper.emitted().next).toBeUndefined();

      wrapper.find('recorder-view-stub').vm.$emit('confirm', recordedFile);

      expect(wrapper.emitted().next).toEqual([[{ recordedFile }]]);
    });

    it('should emit cancel', () => {
      expect.assertions(2);

      expect(wrapper.emitted().cancel).toBeUndefined();

      wrapper.find('recorder-view-stub').vm.$emit('cancel');

      expect(wrapper.emitted().cancel).toEqual([[]]);
    });
  });
});
