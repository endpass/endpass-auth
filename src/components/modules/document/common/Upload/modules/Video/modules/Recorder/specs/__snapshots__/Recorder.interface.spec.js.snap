// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecorderInterface render should be a correct component 1`] = `
<portal-stub name="1" order="0" slotprops="[object Object]" tag="DIV" to="portal-layout">
  <full-screen-stub>
    <recorder-view-stub temp="true"></recorder-view-stub>
  </full-screen-stub>
</portal-stub>
`;
